app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaScreen.kt

```diff
@@ -88,11 +82,39 @@ fun checkIfSharingIsEnabled(featureFlagNotificationViewModel: FeatureFlagNotific
     featureFlagNotificationViewModel.updateSharingStateIfNeeded()
 }
 
+@Composable
+fun ImportMediaContent(
+    unauthorizedViewModel: ImportMediaUnauthorizedViewModel,
+    featureFlagNotificationViewModel: FeatureFlagNotificationViewModel
+) {
+
+    when (featureFlagNotificationViewModel.featureFlagState.fileSharingRestrictedState) {
+        FeatureFlagState.SharingRestrictedState.NO_USER -> ImportMediaLoggedOutContent(
+            featureFlagNotificationViewModel = featureFlagNotificationViewModel,
+            viewModel = unauthorizedViewModel
+        )
+        FeatureFlagState.SharingRestrictedState.RESTRICTED_IN_TEAM -> ImportMediaRestrictedContent(
+            featureFlagNotificationViewModel = featureFlagNotificationViewModel
+        )
+        FeatureFlagState.SharingRestrictedState.NONE -> ImportMediaRegularContent()
+    }
+
+    BackHandler { unauthorizedViewModel.navigateBack() }
```

Comment:
Not sure whether just having this navigate back method, justifies the creation of the `ImportMediaUnauthorizedViewModel` unless we are planning to add more methods there soon for sure.

Couldn't we simply use the `featureFlagNotificationViewModel.navigateBack()` method instead?

Reply:
I think it works the same even without this `BackHandler { unauthorizedViewModel.navigateBack() }` as the navigation handles back clicks for us, as you can see we don't implement the `BackHandler` in every screen 😄 We could remove it along with the `ImportMediaUnauthorizedViewModel` which is just to provide this `navigateBack()`.

Reply:
I checked and it actually also interferes and anticipates the other `BackHandler` when you are logged in and the search is active, then it closes the screen instead of just closing the search, so I'm even more convinced that we should abandon this one.